### ISSUE TITLE
Problem with synchonization with QField

### ISSUE THREAD

**Issue by EgidijusViko** [opengisch/qfieldcloud#873]
Hello,

I have issue with synchronization with project. When I try sync project on android side it show.
"Downloading error. Packining job finished unsuccessfully:..."
```
On QFieldCloud side I see this error
db-1                           | 2024-02-02 07:55:57.304 EET [32] LOG:  statement: BEGIN
db-1                           | 2024-02-02 07:55:57.304 EET [32] LOG:  statement: UPDATE "core_job" SET "docker_finished_at" = '2024-02-02T05:55:57.302785+00:00'::timestamptz WHERE "core_job"."id" = '07f0ef2b-629a-4925-aad6-e96482546c6a'::uuid
db-1                           | 2024-02-02 07:55:57.306 EET [32] LOG:  statement: COMMIT
competent_chatelet exited with code 0
worker_wrapper-1               | {"request_id":"none","ts":"2024-02-02T05:55:57.355471","level":"INFO","name":"worker_wrapper.wrapper","message":"Finished execution with code 0, logs:\n::<<<::9f243971-866a-4270-a0ff-0c6c01fa3f04 Start QGIS Application\n05:55:55.294 root      INFO     Starting QGIS app version 33402 (exported)...\n05:55:56.288 root      INFO     QGIS app started!\n::>>>::9f243971-866a-4270-a0ff-0c6c01fa3f04 2\n::<<<::0241a189-acad-4411-baa3-94a36b464d0b Download Project Directory\n05:55:56.288 root      INFO     Preparing a temporary directory for project files\u2026\n05:55:56.621 root      INFO     Downloading project files\u2026\n05:55:56.786 /usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py INFO     GET http://app:8000/api/v1/files/bcd9dbdf-0307-4a0e-9992-b4109e96fb85/BASTesamosirplanuojamos.gpkg/ got HTTP 500\n::>>>::0241a189-acad-4411-baa3-94a36b464d0b 1\n05:55:56.791 QGSSTDERR WARNING  QFSFileEngine::open: No file name specified\n05:55:56.792 root      INFO     Stopping QGIS app\u2026\n","filename":"wrapper.py","lineno":375,"thread":140389151479616,"source":"worker_wrapper"}
db-1                           | 2024-02-02 07:55:57.358 EET [32] LOG:  statement: BEGIN
db-1                           | 2024-02-02 07:55:57.358 EET [32] LOG:  statement: UPDATE "core_job" SET "output" = '::<<<::9f243971-866a-4270-a0ff-0c6c01fa3f04 Start QGIS Application
db-1                           |        05:55:55.294 root      INFO     Starting QGIS app version 33402 (exported)...
db-1                           |        05:55:56.288 root      INFO     QGIS app started!
db-1                           |        ::>>>::9f243971-866a-4270-a0ff-0c6c01fa3f04 2
db-1                           |        ::<<<::0241a189-acad-4411-baa3-94a36b464d0b Download Project Directory
db-1                           |        05:55:56.288 root      INFO     Preparing a temporary directory for project files…
db-1                           |        05:55:56.621 root      INFO     Downloading project files…
db-1                           |        05:55:56.786 /usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py INFO     GET http://app:8000/api/v1/files/bcd9dbdf-0307-4a0e-9992-b4109e96fb85/BASTesamosirplanuojamos.gpkg/ got HTTP 500
db-1                           |        ::>>>::0241a189-acad-4411-baa3-94a36b464d0b 1
db-1                           |        05:55:56.791 QGSSTDERR WARNING  QFSFileEngine::open: No file name specified
db-1                           |        05:55:56.792 root      INFO     Stopping QGIS app…
db-1                           |        ', "feedback" = '{"error": "Requested \"http://app:8000/api/v1/files/bcd9dbdf-0307-4a0e-9992-b4109e96fb85/BASTesamosirplanuojamos.gpkg/\" and got \"500 Internal Server Error\":\n{\n  \"code\": \"unknown_error\",\n  \"message\": \"QFieldcloud Unknown Error\"\n}", "error_class": "QfcRequestException", "error_stack": ["  File \"/usr/src/app/qfc_worker/utils.py\", line 556, in run_workflow\n    return_values = step.method(**arguments)\n", "  File \"/usr/src/app/qfc_worker/utils.py\", line 232, in download_project\n    client.download_files(\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 565, in download_files\n    raise err\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 545, in download_files\n    self.download_file(\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 616, in download_file\n    resp = self._request(\"GET\", url, stream=True)\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 811, in _request\n    raise QfcRequestException(response) from err\n"], "error_type": "API_INTERNAL_SERVER_ERROR", "feedback_version": "2.0", "outputs": {"start_qgis_app": {}}, "steps": [{"id": "start_qgis_app", "name": "Start QGIS Application", "returns": {}, "stage": 2}, {"id": "download_project_directory", "name": "Download Project Directory", "returns": {}, "stage": 1}, {"id": "qgis_layers_data", "name": "QGIS Layers Data", "returns": {}, "stage": 0}, {"id": "package_project", "name": "Package Project", "returns": {}, "stage": 0}, {"id": "qfield_layer_data", "name": "Packaged Layers Data", "returns": {}, "stage": 0}, {"id": "stop_qgis_app", "name": "Stop QGIS Application", "returns": {}, "stage": 0}, {"id": "upload_packaged_project", "name": "Upload Packaged Project", "returns": {}, "stage": 0}], "workflow_id": "package_project", "workflow_name": "Package Project", "workflow_version": "2.0", "error_origin": "container", "container_exit_code": 0}' WHERE "core_job"."id" = '07f0ef2b-629a-4925-aad6-e96482546c6a'::uuid
```
d

**Comment by suricactus** [comment]
Do you have your own on-premise installation, or you see this on app.qfield.cloud? If on app.qfield.cloud, can you point me to your project?

**Comment by suricactus** [comment]
Check the logs of the app container, it returns error 500 when you trigger your job.

**Comment by EgidijusViko** [comment]
When trigger job any 500 error not appear on app section. Forgot to mention when select online layer issue it become that you can't upload new changes on Qfieldcloud.

**Comment by suricactus** [comment]
Please provide a step by step way to reproduce your error and a sample project.

I am still sure you experience an error 500 on your app container, it is in the logs your have pasted in the first message:

```
> db-1                           |        ', "feedback" = '{"error": "Requested \"http://app:8000/api/v1/files/bcd9dbdf-0307-4a0e-9992-b4109e96fb85/BASTesamosirplanuojamos.gpkg/\" and got \"500 Internal Server Error\":\n{\n  \"code\": \"unknown_error\",\n  \"message\": \"QFieldcloud Unknown Error\"\n}", "error_class": "QfcRequestException", "error_stack": ["  File \"/usr/src/app/qfc_worker/utils.py\", line 556, in run_workflow\n    return_values = step.method(**arguments)\n", "  File \"/usr/src/app/qfc_worker/utils.py\", line 232, in download_project\n    client.download_files(\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 565, in download_files\n    raise err\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 545, in download_files\n    self.download_file(\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 616, in download_file\n    resp = self._request(\"GET\", url, stream=True)\n", "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 811, in _request\n    raise QfcRequestException(response) from err\n"], "error_type": "API_INTERNAL_SERVER_ERROR", "feedback_version": "2.0", "outputs": {"start_qgis_app": {}}, "steps": [{"id": "start_qgis_app", "name": "Start QGIS Application", "returns": {}, "stage": 2}, {"id": "download_project_directory", "name": "Download Project Directory", "returns": {}, "stage": 1}, {"id": "qgis_layers_data", "name": "QGIS Layers Data", "returns": {}, "stage": 0}, {"id": "package_project", "name": "Package Project", "returns": {}, "stage": 0}, {"id": "qfield_layer_data", "name": "Packaged Layers Data", "returns": {}, "stage": 0}, {"id": "stop_qgis_app", "name": "Stop QGIS Application", "returns": {}, "stage": 0}, {"id": "upload_packaged_project", "name": "Upload Packaged Project", "returns": {}, "stage": 0}], "workflow_id": "package_project", "workflow_name": "Package Project", "workflow_version": "2.0", "error_origin": "container", "container_exit_code": 0}' WHERE "core_job"."id" = '07f0ef2b-629a-4925-aad6-e96482546c6a'::uuid
```

**Comment by kachmul2004** [comment]
It seems there is a problem with the 0.20.40 and 0.20.41 releases. I created a self-hosted setup using DEBUG=0 and ENVIRONMENT=production, using this [yml file ](https://github.com/opengisch/qfieldcloud/blob/4b7a199d9803fe6df1ce7161a1b84ecfdcca7076/docker-compose.override.standalone.yml)

However, each time I push a project via QfieldSync, it fails. Here is an example:

`Container id:
9c01c65d7ca0e1f2229ff8a3a37c49c74f2fa1e0bdc9eae5ec4b81b355226306
Project:
[test (c8945976-42c3-4a1a-b153-7bfdc0cdf10e) owner: admin](https://gis.ihelp.rw/admin/core/project/c8945976-42c3-4a1a-b153-7bfdc0cdf10e/change/)
Status:
Failed
Error type:
API_INTERNAL_SERVER_ERROR
Type:
Process QGIS Project File
Created at:
Feb. 8, 2024, 11:22 a.m.
Updated at:
Feb. 8, 2024, 11:22 a.m.
Started at:
Feb. 8, 2024, 11:22 a.m.
Finished at:
-
Docker started at:
Feb. 8, 2024, 11:22 a.m.
Docker finished at:
Feb. 8, 2024, 11:22 a.m.
Output pre:
::<<<::c67a8df3-9fc0-4f09-a301-728de6d60e72 Start QGIS Application
10:22:21.328 root      INFO     Starting QGIS app version 33402 (exported)...
10:22:22.796 root      INFO     QGIS app started!
::>>>::c67a8df3-9fc0-4f09-a301-728de6d60e72 2
::<<<::c1738075-d2b2-4233-b9ad-e5ec514df78f Download Project Directory
10:22:22.811 root      INFO     Preparing a temporary directory for project files…
10:22:23.088 root      INFO     Downloading project files…
10:22:23.235 /usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py INFO     GET http://app:8000/api/v1/files/c8945976-42c3-4a1a-b153-7bfdc0cdf10e/test.qgz/ got HTTP 500
::>>>::c1738075-d2b2-4233-b9ad-e5ec514df78f 1
10:22:23.240 QGSSTDERR WARNING  QFSFileEngine::open: No file name specified
10:22:23.241 root      INFO     Stopping QGIS app…
Feedback pre:
{
  "container_exit_code": 0,
  "error": "Requested \"http://app:8000/api/v1/files/c8945976-42c3-4a1a-b153-7bfdc0cdf10e/test.qgz/\" and got \"500 Internal Server Error\":\n{\n  \"code\": \"unknown_error\",\n  \"message\": \"QFieldcloud Unknown Error\"\n}",
  "error_class": "QfcRequestException",
  "error_origin": "container",
  "error_stack": [
    "  File \"/usr/src/app/qfc_worker/utils.py\", line 556, in run_workflow\n    return_values = step.method(**arguments)\n",
    "  File \"/usr/src/app/qfc_worker/utils.py\", line 232, in download_project\n    client.download_files(\n",
    "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 565, in download_files\n    raise err\n",
    "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 545, in download_files\n    self.download_file(\n",
    "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 616, in download_file\n    resp = self._request(\"GET\", url, stream=True)\n",
    "  File \"/usr/local/lib/python3.10/dist-packages/qfieldcloud_sdk/sdk.py\", line 811, in _request\n    raise QfcRequestException(response) from err\n"
  ],
  "error_type": "API_INTERNAL_SERVER_ERROR",
  "feedback_version": "2.0",
  "outputs": {
    "start_qgis_app": {}
  },
  "steps": [
    {
      "id": "start_qgis_app",
      "name": "Start QGIS Application",
      "returns": {},
      "stage": 2
    },
    {
      "id": "download_project_directory",
      "name": "Download Project Directory",
      "returns": {},
      "stage": 1
    },
    {
      "id": "project_validity_check",
      "name": "Project Validity Check",
      "returns": {},
      "stage": 0
    },
    {
      "id": "opening_check",
      "name": "Opening Check",
      "returns": {},
      "stage": 0
    },
    {
      "id": "project_details",
      "name": "Project Details",
      "returns": {},
      "stage": 0
    },
    {
      "id": "generate_thumbnail_image",
      "name": "Generate Thumbnail Image",
      "returns": {},
      "stage": 0
    },
    {
      "id": "stop_qgis_app",
      "name": "Stop QGIS Application",
      "returns": {},
      "stage": 0
    }
  ],
  "workflow_id": "process_projectfile",
  "workflow_name": "Process Projectfile",
  "workflow_version": "2.0"
}`

**Comment by suricactus** [comment]
May I kindly ask for your app logs? You experience error 500 in the app container?

**Comment by kachmul2004** [comment]
> May I kindly ask for your app logs? You experience error 500 in the app container.

@suricactus 

![Screenshot_20240208_125350.jpg](https://github.com/opengisch/qfieldcloud/assets/34398375/943ab1aa-991f-4822-8307-797a7835b3f6)

**Comment by suricactus** [comment]
I meant the docker compose service called app. Closing this issue as we are not able to support you without relevant information.

**Comment by kachmul2004** [comment]
> I meant the docker compose service called app. Closing this issue as we are not able to support you without relevant information.

Oh, you could have been more specific. Here you go
[app_log.txt](https://github.com/opengisch/qfieldcloud/files/14208176/app_log.txt)

**Comment by suricactus** [comment]
The error says it:
> Expected to either run behind nginx proxy, debug mode or within a test suite.

This is configurable via `QFIELDCLOUD_WORKER_QFIELDCLOUD_URL` https://github.com/opengisch/qfieldcloud/blob/master/.env.example#L96-L97 .

**Comment by Cris70** [comment]
I have the same problem. 
I do not understand what the `QFIELDCLOUD_WORKER_QFIELDCLOUD_URL` configuration directive has to do with this problem.
Can you please elaborate a little, @suricactus ?
Thank you in advance.
Cris

**Comment by suricactus** [comment]
It is a technical requirement for QFieldCloud API to run behind an nginx proxy.

 `QFIELDCLOUD_WORKER_QFIELDCLOUD_URL` sets the URL that `qfieldcloud-sdk` uses to connect to QFieldCloud API from within the `qgis` container.

**Comment by Cris70** [comment]
> It is a technical requirement for QFieldCloud API to run behind an nginx proxy.

Ok, but I still don't get it.
The nginx configuration is already provided in one of the docker images.
Should I add another nginx instance to act as a proxy?
And if so, why it is not explained in the "Deployment" section of the Readme?

**Comment by suricactus** [comment]
> > It is a technical requirement for QFieldCloud API to run behind an nginx proxy.
> 
> Ok, but I still don't get it. The nginx configuration is already provided in one of the docker images.
Yes, this is the one that you need to use.
> Should I add another nginx instance to act as a proxy?
No, it's enough the one that is included in the docker-compose file. If your server runs on `chris70.com`, make sure that `QFIELDCLOUD_WORKER_QFIELDCLOUD_URL=https://chris70.com/api/v1` OR `QFIELDCLOUD_WORKER_QFIELDCLOUD_URL=http://app:8000/api/v1/`. The latter is what I use on my local dev machine.

> And if so, why it is not explained in the "Deployment" section of the Readme?
It might be outdated, I haven't checked it for a while. If you are willing to contribute with improvement in the provided deployment documentation, you are more than welcome on board.

**Comment by Cris70** [comment]
Thank you @suricactus !!
Now it is very clear what I have to do.
I will surely contribute as soon as I test it on my own server!!

**Comment by Cris70** [comment]
Hi @suricactus! Just to let you (and anybody watching this) know: it turned out that I had already made the correct configuration right from the start, but since my server is a VM behind an HAProxy reverse proxy, I had to also made some changes in the HAProxy configuration to make it work correctly.
So there is really nothing that needs to be added to the project's documentation, as this is a very specific use case.
Thank you for your help!
